File: Models/Goal.cs
using System;

namespace Monatsziele.Api.Models
{
    public class Goal
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double CurrentAmount { get; set; }
        public double TargetAmount { get; set; }
        public double CurrentPercentage { get; set; }
        public string AmountUnitOfMeasurement { get; set; }
    }
}

File: src/app/goal.ts
import { Period } from './period.enum';
import { Recording } from './recording';

export class Goal {
    id: string;
    name: string;
    description?: string;
    period: Period;
    currentAmount: number;
    targetAmount: number;
    amountUnitOfMeasurement?: string;
    currentPercentage?: number;
    recordings?: Recording[];
    imageUrl?: string;
    imageSourceUrl?: string;
    imagePhotographerName?: string;
}
